src/misc/setup.cpp

```diff
@@ -674,13 +676,36 @@ const std::vector<Value>& PropMultiVal::GetValues() const
 	return valid_values;
 }
 
+// When setting a property's list of valid values (For example, composite =
+// [auto, on, off]), this function inspects the given valid value to see if it's
+// boolean string (ie: "on" or "off"). If so, this records if a boolean is valid
+// and its direction (either positive or negative) so we can accept all of those
+// corresponding boolean strings from the user (ie: composite = disabled")
+//
+void Property::MaybeSetBoolValid(const std::string_view valid_value)
+{
+	// Deliberately ignore the numerical bools to avoid false-positive
+	if (valid_value == "0" || valid_value == "1") {
+		return;
+	}
```

Comment:
Nice 👍🏻

Reply:
Actually, I'm wondering if this approach is a bit backwards and error prone. How about _only_ "whitelisting" `0` and `1` as valid values in the "pure bool" config param handler in `setup.cpp`, but not in `parse_bool_setting`? So to be clear, `parse_bool_setting` would only handle the textual descriptions and _not_ `0` and `1`.

The reason I'd prefer to do it this way is because this could result in subtle bugs that might go unnoticed for multi-value settings that accept either `off|none|disabled|no` for "disabling" the feature, and an integer for the "enabled" state.

Reply:
Yup - agree that we should snip the numeric aliasing to bool.